### PR TITLE
 Enable eslint recommended rules before google rules and fix existing issues

### PR DESCRIPTION
**Pull request by dstockwell**

### CODE REVIEW

File: runtime/api-channel.js [hunk 1]
```diff
@@ -252,8 +252,6 @@ class PECOuterPort extends APIPort {
     super(messagePort, 'o');
 
     this.registerCall('Stop', {});
-    this.registerCall('DefineParticle',
```
Comment (smalls) [inline runtime/api-channel.js]: This doesn't seem like a lint change - is it an unrelated cl that snuck in?

File: runtime/arc.js [hunk 2]
```diff
@@ -154,7 +154,7 @@ class Arc {
         case 'firebase':
```
Comment (smalls) [inline runtime/arc.js]: Why does this case not have `{}`?

Comment (smalls) [reply runtime/arc.js]: (I assume the rule is 1-line, and it counts 1-line+break as a still 1-line?)

File: runtime/inner-PEC.js [hunk 3]
```diff
@@ -119,11 +118,6 @@ class InnerPEC {
       transformationParticle.renderHostedSlot(transformationSlotName, hostedSlotId, content);
     };
 
-    this._apiPort.onDefineParticle = ({particleDefinition, particleFunction}) => {
```
Comment (smalls) [inline runtime/inner-PEC.js]: Unrelated?

File: runtime/schema.js [hunk 4]
```diff
@@ -153,7 +153,6 @@ class Schema {
           throw new TypeError(
               `Type mismatch ${op}ting field ${name} (union [${fieldType.types}]); ` +
               `value '${value}' is type ${typeof(value)}`);
-          break;
 
         case 'schema-tuple':
```
Comment (smalls) [inline runtime/schema.js]: Do these cases need `{}`?

File: runtime/test/data-layer-test.js [hunk 5]
```diff
@@ -41,6 +41,8 @@ describe('entity', async function() {
 
 describe.skip('relation', function() {
   it('can be created, stored, and restored', function() {
+    let Relation;
```
Comment (smalls) [inline runtime/test/data-layer-test.js]: Won't line 46/47 (`new Relation`) always fail because `Relation` is undefined? This may be a misunderstanding of JS on my part though :)

File: shell/app-shell/elements/arc-auth.js [hunk 6]
```diff
@@ -33,20 +33,17 @@ class ArcAuth extends Xen.Base {
...
-          firebase.auth().getRedirectResult().then(result => {
-            if (result.user === null) {
-              firebase.auth().signInWithRedirect(provider);
-            } else {
-              this._credential(result);
-            }
-          });
-          */
-        }
+        // firebase.auth().signInWithPopup(provider).then(this._credential.bind(this));
```
Comment (smalls) [inline shell/app-shell/elements/arc-auth.js]: It seems as though someone was in-progress on this - any thoughts? If they're done, the commented out code might be ready for removal too.

File: shell/artifacts/Common/source/ShowCollection.js [hunk 7]
```diff
@@ -153,7 +153,7 @@ recipe
       let hostedItems = this._state.hostedItems || [];
       let listIndex = hostedItems.findIndex(item => item.subId == subId);
       let item = Object.assign({}, content.model, {subId});
-      if (listIndex >= 0 && listIndex < items.length) {
+      if (listIndex >= 0 && listIndex < hostedItems.length) {
```
Comment (smalls) [inline shell/artifacts/Common/source/ShowCollection.js]: Was this just failing?

File: runtime/arc.js (hunk 2 above)
Comment (dstockwell) [reply runtime/arc.js]: `{}` is required when a case statement needs a scope to define variables

File: runtime/api-channel.js (hunk 1 above)
Comment (dstockwell) [reply runtime/api-channel.js]: I think a lint error somewhere led me to realize this is unused.

File: runtime/inner-PEC.js (hunk 3 above)
Comment (dstockwell) [reply runtime/inner-PEC.js]: Same as the change in api-channel.js

File: shell/app-shell/elements/arc-auth.js (hunk 6 above)
Comment (dstockwell) [reply shell/app-shell/elements/arc-auth.js]: Maybe @sjmiles knows?

File: runtime/test/data-layer-test.js (hunk 5 above)
Comment (dstockwell) [reply runtime/test/data-layer-test.js]: It would fail, but this test is skipped. Most of the code here has evolved significantly since it was written.

File: runtime/schema.js (hunk 4 above)
Comment (dstockwell) [reply runtime/schema.js]: See earlier comment.

**Comment by dstockwell** [comment]
Merging this now before it rots further :stuck_out_tongue_closed_eyes:
